apps/omg/lib/omg/state/core.ex

```diff
@@ -388,6 +425,44 @@ defmodule OMG.State.Core do
     }
   end
 
+  defp collect_fees(%Core{fees_paid: fees_paid} = state, tx_diffs) do
+    tx_fees =
+      tx_diffs
+      |> Enum.reject(fn {_token, amount} -> amount == 0 end)
+      |> Map.new()
+
+    %Core{
+      state
+      | fees_paid:
+          Map.merge(fees_paid, tx_fees, fn
+            _token, collected, tx_paid -> collected + tx_paid
+          end)
+    }
+  end
+
+  defp disallow_payments(state), do: %Core{state | fee_claiming_started: true}
+
+  defp claim_token(state, token), do: %Core{state | fees_paid: Map.delete(state.fees_paid, token)}
```

Comment:
naming doesn't click for me here. Do `claim_token` and `claim_fees` have the same meaning to the word "claim"? Also more specifically maybe `claim_token` -> `flush_collected_fees_for_token`. A bit longish, but this is what it is